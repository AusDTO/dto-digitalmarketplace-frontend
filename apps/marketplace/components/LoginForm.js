import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import { rootPath } from 'marketplace/routes'
import DocumentTitle from 'react-document-title'

import AUpageAlert from '@gov.au/page-alerts/lib/js/react'
import ErrorBox from 'shared/form/ErrorBox'
import Textfield from 'shared/form/Textfield'
import { required, validEmail, passwordLength } from '../components/validators'
import LoadingButton from './LoadingButton/LoadingButton'
import style from '../../marketplace/main.scss'
import { DMP_LOCKOUT } from '../../marketplace/constants/constants'

const disabledAttribute = {}
if (DMP_LOCKOUT) disabledAttribute.disabled = true
else disabledAttribute.disabled = false

const LoginForm = props => {
  const { model, submitClicked, handleSubmit, currentlySending } = props
  const query = new URLSearchParams(window.location.search)
  const isForcedLogout = query.get('redirected') !== null

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <div className="row">
      <DocumentTitle title="Login - Digital Marketplace">
        <div className="col-sm-push-2 col-sm-8 col-xs-12">
          <article role="main">
            {isForcedLogout && (
              <AUpageAlert as="info" className={`${style.marginBottom2}`}>
                <span>You have been logged out.</span>
              </AUpageAlert>
            )}
            <ErrorBox
              title="There was a problem signing in"
              model={model}
              submitClicked={submitClicked}
              setFocus={setFocus}
            />
            <header className="page-heading page-heading-without-breadcrumb">
              <h1 className="au-display-xl">Sign in to the Marketplace</h1>
              {!DMP_LOCKOUT && (
                <p>
                  New to the Marketplace? <a href={`${rootPath}/signup`}>Create your account</a>
                </p>
              )}
            </header>
            {DMP_LOCKOUT && (
              <AUpageAlert as="error" className={`${style.marginTop2} ${style.marginBottom2}`}>
                <div>
                  <p>
                    Digital Marketplace is closed while it is moving to{` `}
                    <a href="/api/2/r/buyict">BuyICT</a>.
                  </p>
                </div>
              </AUpageAlert>
            )}
            <Form model={model} id="login" onSubmit={data => handleSubmit(data)}>
              <Textfield
                model={`${model}.emailAddress`}
                name="emailAddress"
                id="emailAddress"
                htmlFor="emailAddress"
                label="Email"
                type="email"
                validators={{ required, validEmail }}
                {...disabledAttribute}
                messages={{
                  required: 'Your email is required',
                  validEmail: 'A validly formatted email is required.'
                }}
              />
              <Textfield
                model={`${model}.password`}
                name="password"
                id="password"
                htmlFor="password"
                label="Password"
                type="password"
                description="Your password should be at least 10 characters"
                validators={{ passwordLength }}
                {...disabledAttribute}
                messages={{
                  passwordLength: 'Your password should be at least 10 characters'
                }}
              />
              {!DMP_LOCKOUT && (
                <p className={style.paddingBottom2}>
                  <a href={`${rootPath}/reset-password`}>Forgot your password?</a>
                </p>
              )}
              {currentlySending ? (
                <LoadingButton />
              ) : (
                <input
                  className="au-btn"
                  type="submit"
                  value="Sign in"
                  onClick={submitClicked}
                  {...disabledAttribute}
                />
              )}
            </Form>
          </article>
        </div>
      </DocumentTitle>
    </div>
  )
}

LoginForm.defaultProps = {
  submitClicked: null,
  handleSubmit: null,
  currentlySending: false
}

LoginForm.propTypes = {
  model: PropTypes.string.isRequired,
  submitClicked: PropTypes.func,
  handleSubmit: PropTypes.func,
  currentlySending: PropTypes.bool
}

export default LoginForm
