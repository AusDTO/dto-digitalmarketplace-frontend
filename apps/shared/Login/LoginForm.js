import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import { Link } from 'react-router-dom'

import ErrorBox from '../../shared/form/ErrorBox'
import Textfield from '../../shared/form/Textfield'
import { required, validEmail, passwordLength } from '../../shared/validators'
import LoadingButton from '../LoadingButton/LoadingButton'

const LoginForm = props => {
  const { model, submitClicked, handleSubmit, currentlySending, rootPath, framework } = props

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <div className="row">
      <div className="col-sm-8 col-xs-12">
        <article role="main">
          <ErrorBox
            title="There was a problem signing in"
            model={model}
            submitClicked={submitClicked}
            setFocus={setFocus}
          />
          <header className="page-heading page-heading-without-breadcrumb">
            <h1 className="au-display-xl">
              Sign in to {framework}
            </h1>
          </header>
          <p>
            New to {framework}?{' '}
            <Link to={`${rootPath}/signup`}>
              <strong>Create your account</strong>.
            </Link>
          </p>
          <Form model={model} id="login" onSubmit={data => handleSubmit(data)}>
            <Textfield
              model={`${model}.emailAddress`}
              name="emailAddress"
              id="emailAddress"
              htmlFor="emailAddress"
              label="Email"
              type="email"
              validators={{ required, validEmail }}
              messages={{
                required: 'Your email is required',
                validEmail: 'A validly formatted email is required.'
              }}
            />
            <Textfield
              model={`${model}.password`}
              name="password"
              id="password"
              htmlFor="password"
              label="Password"
              type="password"
              description="Your password should be at least 10 characters"
              validators={{ passwordLength }}
              messages={{
                passwordLength: 'Your password should be at least 10 characters'
              }}
            />
            <p>
              <Link to={`${rootPath}/reset-password`}>
                <strong>Forgot your password?</strong>
              </Link>
            </p>
            <p>
              {currentlySending
                ? <LoadingButton />
                : <input className="au-btn" type="submit" value="Sign in" onClick={submitClicked} />}
            </p>
          </Form>
        </article>
      </div>
    </div>
  )
}

LoginForm.defaultProps = {
  submitClicked: null,
  handleSubmit: null,
  currentlySending: false
}

LoginForm.propTypes = {
  model: PropTypes.string.isRequired,
  submitClicked: PropTypes.func,
  handleSubmit: PropTypes.func,
  currentlySending: PropTypes.bool,
  rootPath: PropTypes.string.isRequired,
  framework: PropTypes.string.isRequired
}

export default LoginForm
