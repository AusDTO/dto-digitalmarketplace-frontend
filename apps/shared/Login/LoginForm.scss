.loginButton {
  padding:  0px
}
