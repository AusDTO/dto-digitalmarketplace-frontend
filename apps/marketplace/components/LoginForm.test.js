import React from 'react'
import expect from 'expect'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { Provider } from 'react-redux'

import configureStore from '../store'
import LoginForm from './LoginForm'

jest.mock('shared/Icon/_getIcons')

describe('Test suite for LoginForm component', () => {
  const state = {
    submitClicked: () => true,
    handleSubmit: () => true,
    model: 'loginForm'
  }

  const store = configureStore()
  test('LoginForm renders without errors', () => {
    const renderer = new ReactShallowRenderer()
    const tree = renderer.render(
      <Provider store={store}>
        <LoginForm {...state} />
      </Provider>
    )

    expect(tree).toMatchSnapshot()
  })
})
