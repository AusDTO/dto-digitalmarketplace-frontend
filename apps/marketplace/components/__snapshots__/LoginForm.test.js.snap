// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for LoginForm component LoginForm renders without errors 1`] = `
<LoginForm
  currentlySending={false}
  handleSubmit={[Function]}
  model="loginForm"
  submitClicked={[Function]}
/>
`;
